pkg/asset/installconfig/gcp/validation.go

```diff
@@ -336,3 +337,86 @@ func validateCredentialMode(client API, ic *types.InstallConfig) field.ErrorList
 
 	return allErrs
 }
+
+func validateMarketplaceImages(client API, ic *types.InstallConfig) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	const errorMessage string = "could not find the boot image: %v"
+
+	if ic.GCP.DefaultMachinePlatform != nil && ic.GCP.DefaultMachinePlatform.OSImage.Name != "" {
+		osImage := ic.GCP.DefaultMachinePlatform.OSImage
+		image, err := client.GetImage(context.TODO(), osImage.Name, osImage.Project)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "gcp", "defaultMachinePlatform", "osImage"), osImage, fmt.Sprintf(errorMessage, err)))
+		} else {
+			errMsg := checkArchitecture(image.Architecture, ic.ControlPlane.Architecture, "controlPlane")
+			if errMsg != "" {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("platform", "gcp", "defaultMachinePlatform", "osImage"), osImage, errMsg))
+			}
```

Comment:
We cannot do this here. As you check in the controlPlane block, `ic.ControlPlane` can be `nil`.
I suggest you move this to the ControlPlane section and if `ic.ControlPlane.Platform.GCP.OSImage.Name == ""` then you use the `image` from the `DefaultMachinePlatform`.